app/controllers/tokens_controller.rb

```diff
@@ -1,16 +1,16 @@
-
- class TokensController < ApplicationController
+class TokensController < ApplicationController
   before_action :set_token, only: [:validate]
   skip_before_action :authenticate_user, only: [:validate]
 
   def validate
-    @user = @token.user
+    @user = @token.get_user
+    @token.try(:destroy)
     @user.update(active: true)
   end
 
   private
     def set_token
-      @token = Token.active.find_by(temp: params[:temp_token]).try(:destroy)
+      @token = Token.active.find_by(temp: params[:temp_token])
```

Comment:
Nice catch @andela-toduah. I can't recall any scenario that will require enforcing the `active` constraint. Moreso since **temp_tokens** are deleted immediately, it's arguable that a token will never be `inactive`. I've made the necessary corrections.